### PR TITLE
Handle server schema version 3

### PR DESCRIPTION
**Pull request by raman325**
Refer to https://github.com/home-assistant-libs/zwave-js-server-python/issues/162 for a reference of what all has been changed

### CODE REVIEW

File: zwave_js_server/model/log_config.py [hunk 1]
```diff
@@ -42,9 +42,9 @@ def to_dict(self) -> LogConfigDataType:
     def from_dict(data: LogConfigDataType) -> "LogConfig":
         """Return LogConfig from LogConfigDataType dict."""
         return LogConfig(
-            data.get("enabled"),
-            data.get("level"),
-            data.get("logToFile"),
-            data.get("filename"),
-            data.get("forceConsole"),
+            data["enabled"],
```
Comment (MartinHjelmare) [inline zwave_js_server/model/log_config.py:45]: Are all keys required now? Then we can update `LogConfigDataType` to have total  True.

File: zwave_js_server/model/node.py [hunk 2]
```diff
@@ -156,34 +163,44 @@ def is_listening(self) -> Optional[bool]:
...
-        """Return the max_baud_rate."""
-        return self.data.get("maxBaudRate")
+    def max_data_rate(self) -> Optional[int]:
+        """Return the max_data_rate."""
+        return self.data.get("maxDataRate")
+
+    @property
+    def supported_data_rates(self) -> Optional[List[int]]:
+        """Return the supported_data_rates."""
+        return self.data.get("supportedDataRates")
```
Comment (MartinHjelmare) [inline zwave_js_server/model/node.py:183]: Containers should always default to empty container.
```suggestion
        return self.data.get("supportedDataRates", [])
```

File: zwave_js_server/model/node.py [hunk 3]
```diff
@@ -156,34 +163,44 @@ def is_listening(self) -> Optional[bool]:
...
     @property
-    def max_baud_rate(self) -> Optional[int]:
-        """Return the max_baud_rate."""
-        return self.data.get("maxBaudRate")
+    def max_data_rate(self) -> Optional[int]:
+        """Return the max_data_rate."""
+        return self.data.get("maxDataRate")
+
+    @property
+    def supported_data_rates(self) -> Optional[List[int]]:
```
Comment (MartinHjelmare) [inline zwave_js_server/model/node.py:181]:
```suggestion
    def supported_data_rates(self) -> List[int]:
```

File: zwave_js_server/model/notification.py [hunk 4]
```diff
@@ -11,19 +11,48 @@
...
+    """Represent a generic notification event data dict type."""
 
     source: Literal["node"]  # required
     event: Literal["notification"]  # required
     nodeId: int  # required
-    notificationLabel: str  # required
+    ccId: int  # required
+
+
+class EntryControlNotificationArgsDataType(TypedDict, total=False):
```
Comment (MartinHjelmare) [inline zwave_js_server/model/notification.py:22]: If  all keys are required we can set total to True.
```suggestion
class EntryControlNotificationArgsDataType(TypedDict, total=True):
```

File: zwave_js_server/model/log_config.py (hunk 1 above)
Comment (raman325) [reply zwave_js_server/model/log_config.py:45]: I don't think they ever were optional

Comment (raman325) [reply zwave_js_server/model/log_config.py:45]: Actually, in the reverse (when we update the log config), we support sending partial parameters, so I don't think we can set `Total` to `False`

File: zwave_js_server/model/notification.py (hunk 4 above)
Comment (raman325) [reply zwave_js_server/model/notification.py:22]: `eventData` is optional https://zwave-js.github.io/node-zwave-js/#/api/node?id=quotnotificationquot. Will update the PR accordingly

**Comment by raman325** [comment]
`eventData` in `EntryControlNotificationArgsDataType` can be a buffer. Should we parse the buffer like we do for values?

File: zwave_js_server/model/notification.py [hunk 5]
```diff
@@ -11,19 +11,48 @@
 
 
 class NotificationDataType(TypedDict, total=False):
```
Comment (MartinHjelmare) [inline zwave_js_server/model/notification.py:13]: Can we change here?

File: zwave_js_server/model/notification.py [hunk 6]
```diff
@@ -34,11 +64,66 @@ def node_id(self) -> int:
...
+    @property
+    def data_type(self) -> int:
+        """Return data type property."""
+        return self.data["args"]["dataType"]
+
+    @property
+    def event_data(self) -> Optional[str]:
+        """Return event data property."""
+        event_data = self.data["args"].get("eventData")
+        return parse_buffer(event_data) if event_data else cast(str, event_data)
```
Comment (MartinHjelmare) [inline zwave_js_server/model/notification.py:129]: Why do we cast to string if `event_data` is falsy? Looks like we're potentially casting `None` to string.

Comment (raman325) [reply zwave_js_server/model/notification.py:129]: good catch

**Comment by raman325** [comment]
damn, that extra line took me out of the patch requirement 🙁  I could switch it back but I think this reads better

**Comment by MartinHjelmare** [comment]
We want to test the entry control CC notification anyway, right?

**Comment by raman325** [comment]
> We want to test the entry control CC notification anyway, right?

Ideally yes. I can fake a payload for it but would be great to get one from an actual device

**Comment by raman325** [comment]
Are we ok with proceeding with this as is and adding the Entry Control CC tests later? I'm not sure that we'll be able to get test data quickly if at all

**Comment by MartinHjelmare** [comment]
What devices produce those notifications? I have a lock.

**Comment by raman325** [comment]
> What devices produce those notifications? I have a lock.

I believe it is Secure Keypad Specific Device Class -> https://www.silabs.com/documents/public/miscellaneous/SDS10242-Z-Wave-Device-Class-Specification.pdf page 80

**Comment by raman325** [comment]
It looks like there are three products that qualify for this:
https://products.z-wavealliance.org/products/2478?selectedFrequencyId=2
https://products.z-wavealliance.org/products/3870?selectedFrequencyId=2
https://products.z-wavealliance.org/products/3575?selectedFrequencyId=-1

**Comment by raman325** [comment]
Thanks goes to @WizKid for providing the test data for the last test

**Review by MartinHjelmare — APPROVED** [review]
Nice!

**Pull request merged**